### PR TITLE
DEPR: Deprecate positional arguments in pivot

### PR DESCRIPTION
**Pull request by phofl**
- [x] xref #30228 (Replace xxxx with the Github issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

cc @mroeschke

### CODE REVIEW

File: pandas/core/frame.py [hunk 1]
```diff
@@ -8580,6 +8580,7 @@ def groupby(
 
     @Substitution("")
     @Appender(_shared_docs["pivot"])
+    @deprecate_nonkeyword_arguments(version=None, allowed_args=["self"])
```
Comment (mroeschke) [inline pandas/core/frame.py:8583]: Specify `version="1.5"` here?

File: pandas/core/reshape/pivot.py [hunk 2]
```diff
@@ -472,6 +473,7 @@ def _convert_by(by):
 
 @Substitution("\ndata : DataFrame")
 @Appender(_shared_docs["pivot"], indents=1)
+@deprecate_nonkeyword_arguments(version=None, allowed_args=["data"])
```
Comment (mroeschke) [inline pandas/core/reshape/pivot.py:476]: Same?

Comment (phofl) [reply pandas/core/reshape/pivot.py:476]: We are always using version=None, not sure why though. So wanted to be consistent here

**Review by mroeschke — APPROVED** [review]
LGTM. Could you merge in main once more? (I'm fairly sure the timeouts are unrelated)

**Comment by phofl** [comment]
Done

**Comment by phofl** [comment]
Green now, merging

**Pull request merged**